test/ability_dry_skin.c

```diff
@@ -0,0 +1,114 @@
+    } SCENE {
+        ABILITY_POPUP(player, ABILITY_DRY_SKIN);
+        HP_BAR(player, damage: TEST_MAX_HP / 8);
+        MESSAGE("The Heliolisk's Dry Skin takes its toll!");
+    }
+}
+
+SINGLE_BATTLE_TEST("Dry Skin heals 1/8th Max HP in Rain")
+{
+    GIVEN {
+        PLAYER(SPECIES_HELIOLISK) { Ability(ABILITY_DRY_SKIN); HP(100); MaxHP(TEST_MAX_HP); };
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_RAIN_DANCE); }
+    } SCENE {
+        ABILITY_POPUP(player, ABILITY_DRY_SKIN);
+        MESSAGE("Heliolisk's Dry Skin restored its HP a little!");
+        HP_BAR(player, hp: TEST_MAX_HP / 8 + 100);
+    }
+}
+
+SINGLE_BATTLE_TEST("Dry Skin increases damage taken from Fire-type moves by 25%", s16 damage)
+{
+    GIVEN {
+        ASSUME(gBattleMoves[MOVE_EMBER].type == TYPE_FIRE);
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_HELIOLISK) { Ability(ABILITY_DRY_SKIN); };
+    } WHEN {
+        TURN {MOVE(player, MOVE_EMBER); }
+    } SCENE {
+        MESSAGE("Wobbuffet used Ember!");
+        HP_BAR(opponent, captureDamage: &results[i].damage);
+    } FINALLY {
+        EXPECT_MUL_EQ(results[0].damage, Q_4_12(1.25), results[1].damage);
+    }
+}
+
+SINGLE_BATTLE_TEST("Dry Skin heals 25% when hit by water type moves")
+{
+    GIVEN {
+        ASSUME(gBattleMoves[MOVE_BUBBLE].type == TYPE_WATER);
+        PLAYER(SPECIES_HELIOLISK) { Ability(ABILITY_DRY_SKIN); HP(100); MaxHP(TEST_MAX_HP); };
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_BUBBLE); }
+    } SCENE {
+        ABILITY_POPUP(player, ABILITY_DRY_SKIN);
+        HP_BAR(player, hp: TEST_MAX_HP / 4 + 100);
+        MESSAGE("Heliolisk restored HP using its Dry Skin!");
+    }
+}
+
+SINGLE_BATTLE_TEST("Dry Skin does not activate if protected")
+{
+    GIVEN {
+        ASSUME(gBattleMoves[MOVE_BUBBLE].type == TYPE_WATER);
+        PLAYER(SPECIES_HELIOLISK) { Ability(ABILITY_DRY_SKIN); HP(100); MaxHP(TEST_MAX_HP); };
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_PROTECT); MOVE(opponent, MOVE_BUBBLE); }
+    } SCENE {
+        NONE_OF { ABILITY_POPUP(player, ABILITY_DRY_SKIN); HP_BAR(player); MESSAGE("Heliolisk restored HP using its Dry Skin!"); }
+    }
+}
+
+SINGLE_BATTLE_TEST("Dry Skin is only triggered once on multi strike moves")
+{
+    GIVEN {
+        ASSUME(gBattleMoves[MOVE_WATER_SHURIKEN].type == TYPE_WATER);
+        ASSUME(gBattleMoves[MOVE_WATER_SHURIKEN].effect == EFFECT_MULTI_HIT);
+        PLAYER(SPECIES_HELIOLISK) { Ability(ABILITY_DRY_SKIN); HP(100); MaxHP(TEST_MAX_HP); };
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_WATER_SHURIKEN); }
+    } SCENE {
+        ABILITY_POPUP(player, ABILITY_DRY_SKIN);
+        HP_BAR(player, hp: TEST_MAX_HP / 4 + 100);
+        MESSAGE("Heliolisk restored HP using its Dry Skin!");
+    }
+}
+
+SINGLE_BATTLE_TEST("Dry Skin prevents Items from activating")
```

Comment:
```suggestion
SINGLE_BATTLE_TEST("Dry Skin prevents Absorb Bulb and Luminous Moss from activating")
```

Reply:
This is to match wording in the test for Water Absorb's identical effect.

`SINGLE_BATTLE_TEST("Water Absorb prevents Items from activating")`

Which wording would you rather have?

Reply:
The second one, because I feel "Items" is too vague to describe what's happening.

Reply:
The one I proposed for both, because I feel "Items" is too vague to describe what's happening.

Reply:
Done.